sql/hive/src/main/scala/org/apache/spark/sql/hive/MetastoreRelation.scala

```diff
@@ -140,7 +145,12 @@ private[hive] case class MetastoreRelation(
           sparkSession.sessionState.conf.defaultSizeInBytes
         })
     }
-  )
+    if (catalogTable.catalogStats.isDefined) {
```

Comment:
We can skip the above computation of `sizeInBytes` if `catalogStats` is defined.